src/arrayMethodJoin.js

```diff
@@ -4,9 +4,30 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const length = this.length;
+    let resultJoin = '';
+    let counter = 0;
+
+    while (counter < length) {
+      if (counter > 0) {
+        resultJoin += `${separator}`;
```

Comment:
no need for extrapolation here
```suggestion
        resultJoin += separator;
```